test/internal/register_internal_actor.js

```diff
@@ -0,0 +1,139 @@
+    let accountProvider;
+    let worker;
+
+    beforeEach(async () => {
+        accountProvider = new utils.AccountProvider(accounts);
+        ({
+            internal,
+            worker,
+            organization,
+        } = await InternalUtils.setupInternal(accountProvider));
+    });
+
+    describe('Negative Tests', async () => {
+        it('Reverts if non-worker address is adding internal actor', async () => {
+            const internalActors = [];
+            internalActors.push(accountProvider.get());
+            const nonWorker = accountProvider.get();
+
+            await utils.expectRevert(internal.registerInternalActor(
+                internalActors,
+                { from: nonWorker },
+            ),
+            'Worker should be registered.',
+            'Only whitelisted workers are allowed to call this method.');
+        });
+    });
+
+    describe('Events', async () => {
+        it('Verify InternalActorRegistered event', async () => {
+            const internalActors = [];
+            internalActors.push(accountProvider.get());
+            internalActors.push(accountProvider.get());
+            internalActors.push(accountProvider.get());
+
+            const transactionResponse = await internal.registerInternalActor(
+                internalActors,
+                { from: worker },
+            );
+
+            const events = Event.decodeTransactionResponse(
+                transactionResponse,
+            );
+
+            Event.assertEqualMulti(events, [{
+                name: 'InternalActorRegistered',
+                args:
+            {
+                _organization: organization,
+                _actor: internalActors[0],
+            },
+            },
+            {
+                name: 'InternalActorRegistered',
+                args:
+              {
+                  _organization: organization,
+                  _actor: internalActors[1],
+              },
+            },
+            {
+                name: 'InternalActorRegistered',
+                args:
+              {
+                  _organization: organization,
+                  _actor: internalActors[2],
+              },
+            },
+            ]);
+        });
+
+        it('Do not register already registered internal actor', async () => {
+            const internalActors = [];
+            internalActors.push(accountProvider.get());
+
+            await internal.registerInternalActor(
+                internalActors,
+                { from: worker },
+            );
+
+            const transactionResponse = await internal.registerInternalActor(
+                internalActors,
+                { from: worker },
+            );
+
+            const events = Event.decodeTransactionResponse(
+                transactionResponse,
+            );
+
+            assert.strictEqual(
+                events.length,
+                0,
+                'Should not emit InternalActorRegistered',
```

Comment:
Suggest updating it to below:

"Should not emit InternalActorRegistered event"